python/SNS_ENCRYPTED_TOPIC_CHECK/SNS_ENCRYPTED_TOPIC_CHECK.py

```diff
@@ -0,0 +1,451 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+'''
+Rule Name:
+  SNS_ENCRYPTED_TOPIC_CHECK
+
+Description:
+  Checks whether the Amazon Simple Notification Service (SNS) topics are encrypted. The rule is NON_COMPLIANT if SNS topic is not encrypted or encrypted with a different key then "KmsKeyId".
+
+Trigger:
+  Periodic
+
+Reports on:
+  AWS::SNS::Topic
+
+Rule Parameters:
+  KmsKeyId
+    (Optional) Comma-separated list of KMS keys ARN that should used to encrypt the Amazon Simple Notification Service
+
+Scenarios:
+  Scenario: 1
+     Given: No SNS topic is present
+      Then: Return NOT_APPLICABLE
+  Scenario: 2
+     Given: KmsKeyId parameter is configured
+       And: KmsKeyId does not contain valid KMS Key ARN(s)
+      Then: Return ERROR
+  Scenario: 3
+     Given: There is at least 1 SNS topic
+       And: SNS topic is not encrypted
+      Then: Return NON_COMPLIANT with Annotation "The Amazon Simple Notification Service topic is not encrypted."
+  Scenario: 4
+     Given: There is at least 1 SNS topic
+       And: SNS topic is encrypted
+       And: KmsKeyId parameter is not configured
+      Then: Return COMPLIANT
+  Scenario: 5
+     Given: There is at least 1 SNS topic
+       And: SNS topic is encrypted
+       And: KmsKeyId parameter is configured
+       And: SNS topic encryption Key not matching any KMS key(s) in KmsKeyId
+      Then: Return NON_COMPLIANT with Annotation "The Amazon Simple Notification Service topic is not encrypted with KMS Key {KmsKeyId}"

```

Comment:
"This SNS topic is not encrypted with KMS Key {KmsKeyId}."

Reply:
updated annotation as requested